playground/eShopLite/CatalogDb/Program.cs

```diff
@@ -1,9 +1,24 @@
 using CatalogDb;
+using Microsoft.EntityFrameworkCore;
+using Npgsql;
 
 var builder = WebApplication.CreateBuilder(args);
 
 builder.AddServiceDefaults();
-builder.AddNpgsqlDbContext<CatalogDbContext>("catalogdb");
+
+// Configure Npgsql.EntityFrameworkCore.PostgreSQL services
+if (builder.Configuration.GetConnectionString("catalogdb") is string connectionString)
+{
+    // a workaround for https://github.com/npgsql/efcore.pg/issues/2821
+    var configureNpgsqlLogging = (NpgsqlDataSourceBuilder builder) => { builder.UseLoggerFactory(null); };
+
+    builder.Services
+        .AddNpgsqlDataSource(connectionString, configureNpgsqlLogging)
+        .AddDbContextPool<CatalogDbContext>(dbContextOptionsBuilder => dbContextOptionsBuilder.UseNpgsql());
+}
+
+// Add the Aspire components for Npgsql.EntityFrameworkCore.PostgreSQL (health-check, tracing, metrics)
+builder.AddNpgsqlEntityFrameworkCore<CatalogDbContext>();
```

Comment:
Does this need to be called **after** adding the `DbContext` to DI?

Reply:
Doesn't matter, you prefer to have it inverted to group the `builder.Add...` vs `builder.Services. ...` ?